src/mods/savest_ui.cpp

```diff
@@ -45,9 +46,22 @@ void tick() {
         s_active_state_slot = cstick_dir;
         draw::notify(draw::WHITE, "Slot %d Selected", cstick_dir + 1);
     }
-    auto& state = s_states[s_active_state_slot];
 
     if (pad::button_pressed(mkb::PAD_BUTTON_X)) {
+        // switch to unused slot if pref is active
+        if (pref::get(pref::BoolPref::SavestateSwitchToUnused)) {
+            // dont override current slot unless all full
+            for (u32 i = 0; i < 8; i++) {
+                auto& state = s_states[(s_active_state_slot + i) % 8];
+                if (state.isEmpty()) {
+                    s_active_state_slot = (s_active_state_slot + i) % 8;
+                    break;
+                }
+            }
+        }
+
+        auto& state = s_states[s_active_state_slot];
+
```

Comment:
Can we have a separate "save slot id" and "load slot id"? Usamune has this as well, should be nice for loading the last saved state but saving to a new slot. You could also have an option for the user manually choose the save slot with Z+C or something, in addition to auto picking an empty slot.